app/helpers/application_helper/button/generic_object_definition_button_button_edit.rb

```diff
@@ -0,0 +1,5 @@
+class ApplicationHelper::Button::GenericObjectDefinitionButtonButtonEdit < ApplicationHelper::Button::Basic
+  def visible?
+    @record.kind_of?(CustomButton)
```

Comment:
this test and the one below are just wrong

you are reverting the flow of login in the buttons and toolbars.

why did the controller request a toolbar with this button if the @record is not there?

it you grep `app/helpers/application_helper` for `kind_of?` you have 1/2 of the hits in GenericObjects

that is not because GO are special, that is because the design here is the other way round :-(

Reply:
```
devil - [~/Projects/manageiq-ui-classic] (gtl_fix_access_control)$ grep -r kind_of app/helpers/application_helper/button/
app/helpers/application_helper/button/generic_object_definition_button_button_group_delete.rb:    @record.kind_of?(CustomButtonSet)
app/helpers/application_helper/button/generic_object_definition_button_button_group_delete.rb:    !@record.kind_of?(CustomButtonSet) || (@record.kind_of?(CustomButtonSet) && !@record.custom_buttons.count.zero?)
app/helpers/application_helper/button/mixins/active_context_mixin.rb:    if tree.kind_of?(Array)
app/helpers/application_helper/button/mixins/active_context_mixin.rb:    if tab.kind_of?(Array)
app/helpers/application_helper/button/host_feature_button.rb:      if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
app/helpers/application_helper/button/host_feature_button.rb:      return @record.is_available?(@feature) if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager::Host)
app/helpers/application_helper/button/instance_check_compare.rb:    !@record.kind_of?(OrchestrationStack) || @display != 'instances'
app/helpers/application_helper/button/generic_object_definition_button_button_group_edit.rb:    @record.kind_of?(CustomButtonSet)
app/helpers/application_helper/button/topology_feature_button.rb:    return false if @record.kind_of?(ManageIQ::Providers::InfraManager)
app/helpers/application_helper/button/host_feature_button_with_disable.rb:      return false if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
app/helpers/application_helper/button/host_feature_button_with_disable.rb:      return @record.is_available?(@feature) if @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager::Host)
app/helpers/application_helper/button/generic_object_definition_button_button_group_new.rb:    @actions_node || @record.kind_of?(GenericObjectDefinition)
app/helpers/application_helper/button/vm_snapshot_revert.rb:    return false if @record.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm)
app/helpers/application_helper/button/generic_object_definition_button_delete.rb:    @record.kind_of?(GenericObjectDefinition) && @display != 'generic_objects' && !@actions_node
app/helpers/application_helper/button/generic_object_definition_button_delete.rb:    @record.kind_of?(GenericObjectDefinition) && !@record.generic_objects.count.zero?
app/helpers/application_helper/button/generic_object_definition_button_edit.rb:    !(@display == 'generic_objects' || !@record.kind_of?(GenericObjectDefinition) || @actions_node)
```

Reply:
Ok, will try and get rid of the `kind_of?'s`.

And would have to create toolbars for the individual types of entities instead.

Reply:
Addressed in b41a559